Add sha256 & string to doxygen & cppcheck meta.yml

FarnazH opened the pull request:
A few minor updates:
1. Add `sha256` to meta.yml of cppcheck and doxygen. 
2. Add  `string` option to meta.yml of cppcheck and doxygen, updated packages on <https://anaconda.org/theochem/repo> so that the tar filenames look nicer.
3. Add a short summary for doxygen.

tovrstra commented on tools/cppcheck-conda-recipe/meta.yaml:
```diff
@@ -3,10 +3,12 @@ package:
 
 build:
   number: 3
+  string: 1
```
I guess the purpose is to keep the `string` and `number` identical, 4 in this case. The cppcheck recipe has undergone already some changes, including yours, it becomes 4.

tovrstra commented on tools/doxygen-conda-recipe/meta.yaml:
```diff
@@ -6,11 +6,13 @@ package:
 
 build:
   number: 1
+  string: 1
```
idem (2)

FarnazH commented:
@tovrstra thanks for your comments. I changed the numbers based on your comments, please let me know if there are any mistakes. I updated the files on <https://anaconda.org/theochem/repo>, but unfortunately, I have deleted the earlier builds (with smaller `number` and `string` values). I hope that is fine, as there were no major changes in the recipes. Please let me know.

tovrstra commented:
Looks good. Getting rid of the old files is fine. The old version are not very useful to keep in this case. It would only be different for historical reasons or to guarantee reproducibility of some results. That's certainly not the case here. Merging...
